cloud-controller-manager/do/loadbalancers.go

```diff
@@ -823,14 +841,23 @@ func buildForwardingRules(service *v1.Service) ([]godo.ForwardingRule, error) {
 		protocol := defaultProtocol
 		// Set protocols explicitly if correspondingly configured ports
 		// are found.
+		if udpPortMap[port.Port] {
+			protocol = protocolUDP
+		}
+
+		// we have to delete the ports after we add them to prevent
+		// them from overriding a port that is found in the udp port map
 		if httpPortMap[port.Port] {
 			protocol = protocolHTTP
+			delete(httpPortMap, port.Port)
```

Comment:
> Assuming we want to allow same-port-different-protocol usage: do we know if Kubernetes supports that cleanly? Looking the issue you linked and other issues linked from that, I didn't get a super clear picture on that second point.
Sorry about the confusion on this. I had initially thought that Kubernetes didn't support having the same port for two protocols {TCP, UDP}. But I was a bit wrong about this. The discussion for the actual issue is here:
https://github.com/kubernetes/kubernetes/issues/39188
It seems the TLDR; is that as long as the ports are named, setting the service up for the first time shouldn't be an issue. But it appears due to an issue on K8s side, if the ports are updated, it can end up updating the wrong port/protocol pair.

> And do we (want to) allow reusing the same port for TCP and UDP protocols at the same time from a DO LB perspective? Removing the port here makes me think we don't but a comment further above explains why we don't consider UDP ports in duplicate detection for reuse purposes, so I'm somewhat confused. (Chances are I'm misunderstanding something.)
On the LB side we have support for allowing the same port for TCP and UDP. The deleting ports from the other maps was a way of achieving this behavior. I ran into a problem in the tests where the protocols would end up getting overridden. Since we have this pattern, where if we find a particular port in a particular map (where each map has ports for a particular protocol) we determine that port to be of that protocol. If I don't delete a port once it's found from the maps excluding UDP and that port is found again we end up overwriting its protocol. 
This results in test failure like this:
```
FAIL: Test_buildForwardingRules (0.00s)
    --- FAIL: Test_buildForwardingRules/UDP_and_HTTP2_ports_shared (0.00s)
        loadbalancers_test.go:2203: unexpected forwarding rules
        loadbalancers_test.go:2204: expected: [godo.ForwardingRule{EntryProtocol:"http2", EntryPort:4443, TargetProtocol:"http2", TargetPort:10443, CertificateID:"", TlsPassthrough:true} godo.ForwardingRule{EntryProtocol:"http2", EntryPort:443, TargetProtocol:"http2", TargetPort:10444, CertificateID:"", TlsPassthrough:true} godo.ForwardingRule{EntryProtocol:"udp", EntryPort:4443, TargetProtocol:"udp", TargetPort:10445, CertificateID:"", TlsPassthrough:false} godo.ForwardingRule{EntryProtocol:"udp", EntryPort:443, TargetProtocol:"udp", TargetPort:10446, CertificateID:"", TlsPassthrough:false}]
        loadbalancers_test.go:2205: actual: [godo.ForwardingRule{EntryProtocol:"http2", EntryPort:4443, TargetProtocol:"http2", TargetPort:10443, CertificateID:"", TlsPassthrough:true} godo.ForwardingRule{EntryProtocol:"http2", EntryPort:443, TargetProtocol:"http2", TargetPort:10444, CertificateID:"", TlsPassthrough:true} godo.ForwardingRule{EntryProtocol:"http2", EntryPort:4443, TargetProtocol:"http2", TargetPort:10445, CertificateID:"", TlsPassthrough:true} godo.ForwardingRule{EntryProtocol:"http2", EntryPort:443, TargetProtocol:"http2", TargetPort:10446, CertificateID:"", TlsPassthrough:true}]
```
 I'm happy to change this to a better way if there is one, but that deleting from the map trick is what I came up with.

Reply:
Looks like removing `annDOUDPPorts` like suggessted below would get rid of this change, but first I'll have to rework a large portion of the PR which I am doing now

Reply:
Okay since the annotations are removed, we are just using the Protocol from the service definition and there is no need for this terrible map deletion logic. Thanks @timoreimann!